File: android/src/main/java/org/ligi/passandroid/ui/PassAndroidActivity.java
package org.ligi.passandroid.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import javax.inject.Inject;
import org.greenrobot.eventbus.EventBus;
import org.ligi.passandroid.App;
import org.ligi.passandroid.Tracker;
import org.ligi.passandroid.model.PassStore;
import org.ligi.passandroid.model.Settings;

public class PassAndroidActivity extends AppCompatActivity {

    @Inject
    PassStore passStore;

    @Inject
    Settings settings;

    @Inject
    EventBus bus;

    @Inject
    protected Tracker tracker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        App.component().inject(this);
    }

}

File: android/src/main/java/org/ligi/passandroid/ui/PassAndroidActivity.kt
package org.ligi.passandroid.ui

import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import org.koin.android.ext.android.inject
import org.ligi.passandroid.Tracker
import org.ligi.passandroid.model.PassStore
import org.ligi.passandroid.model.Settings

open class PassAndroidActivity : AppCompatActivity() {

    val passStore: PassStore by inject()
    val settings: Settings by inject()
    val tracker: Tracker by inject()

    private var lastSetNightMode: Int? = null

    override fun onResume() {
        super.onResume()

        if (lastSetNightMode != null && lastSetNightMode != settings.getNightMode()) {
            ActivityCompat.recreate(this)
        }
        lastSetNightMode = settings.getNightMode()
    }

}